data/mllib/pic_data.txt

```diff
@@ -0,0 +1,299 @@
+1000	0.000000	0.000125	0.000038	0.012684	0.000638	0.051091	0.000151	0.044208	0.004264	0.000617	0.007746	0.036569	0.001813	0.000305	0.003171	0.004114	0.000530	0.016800	0.003396	0.017566	0.034756	0.000018	0.051096	0.000022	0.001749	0.000210	0.006065	0.006969	0.016719	0.006028	0.003378	0.003200	0.025072	0.000291	0.000116	0.001633	0.000028	0.011305	0.000019	0.010359	0.006533	0.047593	0.027411	0.000059	0.017558	0.000518	0.000946	0.044212	0.000094	0.005404	0.026762	0.009941	0.003801	0.000027	0.000161	0.000901	0.000019	0.000518	0.034732	0.000059	0.000126	0.000970	0.011814	0.005997	0.000205	0.001832	0.008792	0.036318	0.000149	0.032781	0.010692	0.000530	0.010557	0.016641	0.008180	0.001606	0.000092	0.007445	0.026718	0.027457	0.000957	0.005901	0.000314	0.000162	0.000856	0.004776	0.008114	0.003693	0.000038	0.024965	0.044256	0.007180	0.000022	0.010297	0.000994	0.044255	0.001725	0.016541	0.003658	0.000288
```

Comment:
This is too large for unit tests. Unit tests should be as minimal as possible. For this one, we can construct a very small graph, compute its eigenvector, and derive the clustering result manually, then verify PIC result. For example

~~~
a - b - c - g - h
| \     |   | \ |
d - e - f   i - j
~~~

Assign each edge distance `1` and run PIC with k = 2. The solution should be clear.